### PR TITLE
Ams classification vehicle

### PR DESCRIPTION
**Pull request by rasapala**
Example JSON from test with 2 tags of highest probability:

{'type': 'classification', 'subtype': 'vehicle-attributes', 'classifications': [{'tag': {'value': 'red', '
confidence': 0.5460447072982788}, 'attributes': [{'name': 'color', 'value': 'white', 'confidence': 0.04804454743862152}
, {'name': 'color', 'value': 'gray', 'confidence': 0.16620293259620667}, {'name': 'color', 'value': 'yellow', 'confiden
ce': 0.04688253253698349}, {'name': 'color', 'value': 'red', 'confidence': 0.5460447072982788}, {'name': 'color', 'valu
e': 'green', 'confidence': 0.044175755232572556}, {'name': 'color', 'value': 'blue', 'confidence': 0.015048854053020477
}, {'name': 'color', 'value': 'black', 'confidence': 0.133600652217865}]}, {'tag': {'value': 'truck', 'confidence': 0.9
425309896469116}, 'attributes': [{'name': 'type', 'value': 'car', 'confidence': 0.015209668315947056}, {'name': 'type',
 'value': 'van', 'confidence': 0.004823816474527121}, {'name': 'type', 'value': 'truck', 'confidence': 0.94253098964691
16}, {'name': 'type', 'value': 'bus', 'confidence': 0.037435464560985565}]}]}

fst_highest_prob_class:red
snd_highest_prob_class:truck

### CODE REVIEW

File: extras/ams_wrapper/src/api/models/vehicle_attributes_model.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+from api.models.model import Model
+from api.types import Tag, Attribute, SingleClassification, Classification
+
+logger = get_logger(__name__)
+
+
+
+class VehicleAttributes(Model):
+
+    def load_default_labels(self):
```
Comment (PatrykMatyjasek) [inline extras/ams_wrapper/src/api/models/vehicle_attributes_model.py:29]: maybe this method should be a part of some commons.py?

File: tests/functional/test_single_model_vehicle_attributes.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+from ie_serving.models.models_utils import ModelVersionState, ErrorCode, \
+    _ERROR_MESSAGE  # noqa
+
+
+class TestVehicleAttributes():
+
+    def load_image(self, file_path, width, height):
+        img = cv2.imread(file_path)  # BGR color format, shape HWC
+        #img = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
```
Comment (PatrykMatyjasek) [inline tests/functional/test_single_model_vehicle_attributes.py:37]: commented out code

File: extras/ams_wrapper/src/api/models/vehicle_attributes_model.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        labels_path = os.path.abspath(__file__).replace(".py",".json")
+        try:                                                                          
+            with open(labels_path, 'r') as labels_file:                               
+                data = json.load(labels_file)                                         
+                self.labels = data
+        except Exception as e:                                                        
+            logger.exception("Error occurred while opening labels file: {}".format(e))
```
Comment (PatrykMatyjasek) [inline extras/ams_wrapper/src/api/models/vehicle_attributes_model.py:36]: this message could be confusing, there are two possible errors - unable to open file (permission denied or file does not exist) and invalid file content. I think it should be handled as two (or three) exceptions

File: extras/ams_wrapper/src/api/models/vehicle_attributes_model.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        # model with output shape for color (1,7,1,1) 
+        # with second dimension containing colors
+        # [white, gray, yellow, red, green, blue, black]
+        # model with output shape for type (1,4,1,1) 
+        # with second dimension containing types
+        # [car, bus, truck, van]
+        outputs = self.labels 
+        classifications = []
+        for output in outputs.keys():
+            type_name = output
```
Comment (PatrykMatyjasek) [inline extras/ams_wrapper/src/api/models/vehicle_attributes_model.py:52]: is this assignment needed?

File: extras/ams_wrapper/src/api/models/vehicle_attributes_model.py (hunk 1 above)
Comment (rasapala) [reply extras/ams_wrapper/src/api/models/vehicle_attributes_model.py:29]: Not needed anymore.

File: tests/functional/test_single_model_vehicle_attributes.py (hunk 2 above)
Comment (rasapala) [reply tests/functional/test_single_model_vehicle_attributes.py:37]: Done.

File: extras/ams_wrapper/src/api/models/vehicle_attributes_model.py (hunk 3 above)
Comment (rasapala) [reply extras/ams_wrapper/src/api/models/vehicle_attributes_model.py:36]: Not needed anymore.

File: extras/ams_wrapper/src/api/types.py [hunk 5]
```diff
@@ -74,16 +74,13 @@ def as_dict(self):
         }
 
 class SingleClassification:
-    def __init__(self, tag: Tag, attributes: List[Attribute] = None):
-        self.tag = tag
+    def __init__(self, attributes: List[Attribute] = None):
```
Comment (mzegla) [inline extras/ams_wrapper/src/api/types.py:77]: If we remove tag and classification consists of attributes then we shouldn't allow `None`. 
We would have "empty" object then.

File: extras/ams_wrapper/src/api/types.py [hunk 6]
```diff
@@ -74,16 +74,13 @@ def as_dict(self):
...
         self.attributes = attributes
 
     def as_dict(self):
-        result_dict = {
-            "tag": self.tag.as_dict(),
-        }
+        result_dict = {}
         if self.attributes is not None:
-            result_dict[attributes] = [attribute.as_dict() for attribute in self.attributes]
+            result_dict["attributes"] = [attribute.as_dict() for attribute in self.attributes]
```
Comment (mzegla) [inline extras/ams_wrapper/src/api/types.py:83]: I believe these quotation marks are also required for entity type. Could you also fix it there?

File: extras/ams_wrapper/tests/test_classification_type.py [hunk 7]
```diff
@@ -16,43 +16,43 @@
...
-
-    tag = Tag("car", 0.97)
-    single_classification = SingleClassification(tag)
+    attribute = Attribute("type", "car", 0.97) 
+    attribs = []
+    attribs.append(attribute)
+    single_classification = SingleClassification(attribs)
     assert expected_dict == single_classification.as_dict()
 
 def test_entity():
```
Comment (mzegla) [inline extras/ams_wrapper/tests/test_classification_type.py:35]: You could change name here to `test_classification`

File: tests/functional/test_single_model_vehicle_attributes.py [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+            if color_prob > highest_prob:
+                print("update " + str(color_prob)+ ">" +str(highest_prob))
+                highest_prob = color_prob
+                color_index = i
+
+        print("color_index= " + str(color_index))
+        assert color_index == expected_color_index
+
+
+    def test_run_inference_postprocess(self, vehicle_attributes_model_downloader,
```
Comment (mzegla) [inline tests/functional/test_single_model_vehicle_attributes.py:182]: I don't think this is the right way to test it. 
Post processing itself could be tested via unit tests. 
It shall be functionally tested in a similar manner to:
https://github.com/openvinotoolkit/model_server/blob/ams/tests/functional/test_ams_inference.py
If think you can remove this test for now

File: Jenkinsfile [hunk 9]
```diff
@@ -32,6 +32,12 @@ pipeline {
                         sh './tests/scripts/functional-tests-ams.sh'
                     }
                 }
+
+                stage('functional tests model posprocessing') {
```
Comment (mzegla) [inline Jenkinsfile:36]: Why another stage? Can't we do such things along with other functional tests for AMS?

File: extras/ams_wrapper/src/api/types.py (hunk 5 above)
Comment (rasapala) [reply extras/ams_wrapper/src/api/types.py:77]: Done.

File: extras/ams_wrapper/src/api/types.py (hunk 6 above)
Comment (rasapala) [reply extras/ams_wrapper/src/api/types.py:83]: Done.

File: extras/ams_wrapper/tests/test_classification_type.py (hunk 7 above)
Comment (rasapala) [reply extras/ams_wrapper/tests/test_classification_type.py:35]: Done.

File: tests/functional/test_single_model_vehicle_attributes.py (hunk 8 above)
Comment (rasapala) [reply tests/functional/test_single_model_vehicle_attributes.py:182]: Next step will be to add functional end to end tests.

File: Jenkinsfile (hunk 9 above)
Comment (rasapala) [reply Jenkinsfile:36]: This is postprocessing functional testing not whole end to end. Lets put it in different stage.

**Pull request merged**